File: tests/enums/field-requires-copy.js
function Position$copy(source) {
  return source[0] === 2 ? [2, {
    x: source[1].x,
    y: source[1].y
  }, source[2]] : source[0] === 1 ? [1, {
    x: source[1].x,
    y: source[1].y
  }] : source.slice();
}

export function makeCopy$ofPosition$(position) {
  let copy;
  copy = Position$copy(position);
  return copy;
}
export function makeCopyDirect$ofPosition$(position) {
  return Position$copy(position);
}

File: tests/enums/field-requires-copy.swift
public struct Point {
	var x: Double
	var y: Double
	public init(x _x: Double, y _y: Double) {
		x = _x
		y = _y
	}
}

public enum Position {
	case empty
	case twoDimensional(Point)
	case threeDimensional(Point, Double)
}

public func makeCopy(ofPosition position: Position) -> Position {
	let copy = position
	return copy
}
